util/network-devp2p/src/host.rs

```diff
@@ -672,21 +672,18 @@ impl Host {
 		let nonce = self.info.write().next_nonce();
 		let mut sessions = self.sessions.write();
 
-		let token = sessions.insert_with_opt(|token| {
-			trace!(target: "network", "{}: Initiating session {:?}", token, id);
-			match Session::new(io, socket, token, id, &nonce, &self.info.read()) {
-				Ok(s) => Some(Arc::new(Mutex::new(s))),
-				Err(e) => {
-					debug!(target: "network", "Session create error: {:?}", e);
-					None
-				}
-			}
-		});
+		let entry = sessions.vacant_entry();
+		let key = entry.key();
```

Comment:
is it possible to move these vars under the `Ok(session) =>` match below?

Reply:
ah, I see it's used in the `trace!` now, it's fine as is then

Reply:
Thank you for careful review! That are *literately* my first lines of code written in Rust. :)